modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/provider/test/APIApplicationProviderTest.java

```diff
@@ -142,6 +143,35 @@ public void test() throws Exception {
 		Assert.assertEquals("name", property.getName());
 		Assert.assertEquals(
 			APIApplication.Property.Type.PICKLIST, property.getType());
+
+		// Filter
+
+		HTTPTestUtil.invokeToJSONObject(
```

Comment:
I think it would be simpler to add the filter in the original POST. And maybe have a separate test that tests the things that can be null? The first one would POST a complete full application, and the second one the minimal app with everything that can be null as null. 

I am not asking you to change it, it just crossed my mind. This is ok.

Reply:
As discussed offline, we can leave it as it is. We will refactor the APIApplication creation code with a builder or something similar in the next PRs.